### DISCUSSION TITLE
Übersetzung von Entitätsnamen

### DISCUSSION THREAD

**Discussion by SukramJ** [SukramJ/aiohomematic#1016] [Show and tell]
Mit der Version 2023.4 führt Home Assistant die Möglichkeit zur Übersetzung der Entitätsnamen ein.
Das bedeutet mit Bezug auf diese Integration, dass die bisher nicht übersetzten Parameternamen übersetzt werden (können).

Bsp.: Humidity --> Luftfeuchtigkeit

Einige Parameter habe ich dabei, aufgrund ihrer Systemrelevanz, nicht übersetzt:
- Duty Cycle
- Carrier Sense
- RSSI Peer
- RSSI Device
- ...

Bei einigen Parametern werden die Übersetzungen leer gelassen, um so die Übersichtlichkeit in der UI zu verbessern:
- STATE
- STATUS

Eine vollständige Liste der Übersetzungen findet ihr im [Code](https://github.com/danielperna84/custom_homematic/blob/devel/custom_components/homematicip_local/translations/de.json).

Die Änderungen können ab Version 1.34.0 ausprobiert werden.
Bei bestehenden Installationen, und sofern nicht selbst Hand an die Namen gelegt wurde, werden nur die angezeigten Namen geändert.
Bei einer Neuinstallation und bei neu hinzukommenden Geräten wird auch die `entity_id` übersetzt.

Wäre gut wenn ihr da noch mal mit drauf schauen könntet.

**Comment by patman15** [comment]
Gibt es eine Möglichkeit die Übersetzung der `entity_id` zu deaktivieren? Wenn ich in meinem System alle z.B. Feuchtigkeitssensoren suche (von diversen Integrationen), dann wäre es praktisch weiterhin mit __humidity_ suchen zu können. Zugegeben, mir fallen nicht viele Anwendungen ein, außer Datenbankmigrationen (wie bei mir aktuell bei der Migration der Integration).

**Reply by SukramJ** [reply]
Nein, das ist möglich. Änderungswünsche dazu bitte an HA Core und nicht hier.
Es wird die zum Zeitpunkt der Entitiy ID Generierung eingestellte Sprache verwendet.

**Reply by emufan** [reply]
> Es wird die zum Zeitpunkt der Entitiy ID Generierung eingestellte Sprache verwendet.

Da ich gleichen usecase habe beim Hinzufügen von neuen HM-Geräten, werde/wollte ich das beim nächsten Mal probieren. D.h. auf Englisch stellen, damit es auch bei den neuen Geräten einheitlich wie bei den anderen bei _voltage, _current, _power, etc bleibt und ich weniger Umbenennen muss:

Kennt jemand Nebenwirkungen von temporärem Umstellen der Sprache, d.h. das HA dann wer weiß was auf der DB veranstaltet, andere Sachen umbenennt, usw?

**Reply by SukramJ** [reply]
Mir ist da bisher nichts aufgefallen.
